### ISSUE TITLE
[Bug]: External (http) images and figures fail badly

### ISSUE THREAD

**Issue by Amourspirit** [vscode-restructuredtext/vscode-restructuredtext#416]
### Expected behavior

I expected reStructuredText to handle figure and images that have an external source.

### Actual behavior

But when an external image is used such as :
![sample](https://user-images.githubusercontent.com/4193389/205182487-b1796a72-ec04-4bdc-8a8c-26acdf72039e.png)

Errors are rasise every time the document is edited. I mean each time afte you stop typing.
Renders plugin completly useless.

See Also [RST Scratch project](https://github.com/Amourspirit/python_rst_scratch/tree/go_figure) `go_figure` branch.

Demonstration media.

https://user-images.githubusercontent.com/4193389/205329823-940f32be-661d-4d76-a68b-01651021866b.mp4

### Detailed steps

The steps just add an image or a figure with a source such as

```rst
.. figure:: https://user-images.githubusercontent.com/4193389/205182487-b1796a72-ec04-4bdc-8a8c-26acdf72039e.png
    :alt: The Sheet of Data in SelectListener

    :Sheet Data https image.
```

### Extension information

v189.2.0

### Visual Studio Code information

1.73.1

### Operating system information

Window 10

### Python information

3.9.7

### Sphinx information

5.3.0

### Contents of conf.py

```python
# Configuration file for the Sphinx documentation builder.
#
# For the full list of built-in configuration values, see the documentation:
# https://www.sphinx-doc.org/en/master/usage/configuration.html

# -- Project information -----------------------------------------------------
# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information

import os
import sys
from pathlib import Path

_DOCS_PATH = Path(__file__).parent
_ROOT_PATH = _DOCS_PATH.parent
sys.path.insert(0, str(_ROOT_PATH))

project = 'RST Scratch'
copyright = '2022, Scratch Editor'
author = 'Scratch Editor'
release = '0.1.0'

# -- General configuration ---------------------------------------------------
# https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration

extensions = [
    "sphinx_rtd_theme",
    "sphinx_rtd_dark_mode",
    "sphinx.ext.autodoc",
    "sphinx.ext.viewcode",
    "sphinx_toolbox.collapse",
    "sphinx_toolbox.more_autodoc.autonamedtuple",
    "sphinx_toolbox.more_autodoc.typevars",
    "sphinx_toolbox.more_autodoc.autoprotocol",
    "sphinx.ext.napoleon",
    "sphinx.ext.todo",
    "sphinx.ext.extlinks",
    "sphinx.ext.intersphinx",
    "sphinx_autodoc_typehints",
    "sphinx_tabs.tabs",
    "sphinx_design",
    "sphinxcontrib.spelling",
]

# "sphinx.ext.linkcode",
# sphinx_tabs.tabs docs: https://sphinx-tabs.readthedocs.io/en/latest/

# "sphinx_design"
# https://sphinx-design.readthedocs.io/en/rtd-theme/get_started.html

# region spelling
# https://sphinxcontrib-spelling.readthedocs.io/en/latest/


def get_spell_dictionaries() -> list:

    p = _DOCS_PATH.absolute().resolve() / "internal" / "dict"
    dict_gen = p.glob("spelling_*.*")
    return [str(d) for d in dict_gen if d.is_file()]


spelling_word_list_filename = get_spell_dictionaries()

spelling_show_suggestions = True
spelling_ignore_pypi_package_names = True
spelling_ignore_contributor_names = True
spelling_ignore_acronyms = True

# https://sphinxcontrib-spelling.readthedocs.io/en/latest/customize.html
spelling_exclude_patterns = [".venv/"]

# spell checking;
#   run sphinx-build -b spelling . _build
#       this will checkfor any spelling and create folders with *.spelling files if there are errors.
#       open each *.spelling file and find any spelling errors and fix them in corrsponding files.
#
# spelling_book.txt contains all spelling exceptions related to book in /docs/odev
# spelling_code.txt contains all spelling exceptions related to python doc strings.

# endregion spelling

templates_path = ['_templates']
exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']



# -- Options for HTML output -------------------------------------------------
# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output

# html_theme = 'alabaster'
html_theme = "sphinx_rtd_theme"

# Add any paths that contain custom static files (such as style sheets) here,
# relative to this directory. They are copied after the builtin static files,
# so a file named "default.css" will overwrite the builtin "default.css".
html_static_path = ["_static"]

html_css_files = ["https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.2.1/css/all.min.css"]
if html_theme == "sphinx_rtd_theme":
    html_css_files.append("css/readthedocs_custom.css")

if "sphinx_rtd_dark_mode" in extensions:
    html_css_files.append("css/readthedocs_dark.css")

html_css_files.append("css/style_custom.css")

html_js_files = [
    "js/custom.js",
]

# Napoleon settings
# https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html
napoleon_google_docstring = True
napoleon_include_init_with_doc = True

# https://fossies.org/linux/Sphinx/doc/usage/extensions/autodoc.rst
# This value controls how to represent typehints. The setting takes the following values:
autodoc_typehints = "description"

# see: https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_mock_imports
# see: https://read-the-docs.readthedocs.io/en/latest/faq.html#i-get-import-errors-on-libraries-that-depend-on-c-modules
autodoc_mock_imports = []

# autodoc_type_aliases
# The key of each entry is the type hint as written in the source,
# the value is how it will be written in the generated documentation.
# autodoc_type_aliases = {
#     "Column": "Column",
#     "DictRow": "DictRow",
#     "DocOrCursor": "DocOrCursor",
#     "DocOrText": "DocOrText",
#     "FloatList": "FloatList",
#     "FloatTable": "FloatTable"
# }
autodoc_type_aliases = {}
autodoc_typehints_format = "short"

# https://stackoverflow.com/questions/9698702/how-do-i-create-a-global-role-roles-in-sphinx
# custom global roles or any other rst to include

# sphinx includes s5defs.txt that has baked in roles but must be included.
# style_custom.css contains the colors that match these roles
# https://stackoverflow.com/questions/3702865/sphinx-restructuredtext-set-color-for-a-single-word/60991308#60991308

rst_prolog_lst = [
    ".. include:: <s5defs.txt>",
    "",
]
# rst_prolog_lst.append(f".. |app_ver| replace:: {__version__}\n")

# add extra roles for custom theme colors.
# unlike the colors style_custom.css, these colors can be change by
# changing colors css vars that start with -t-color-
rst_prolog = ""
if html_theme == "sphinx_rtd_theme":
    with open("roles/theme_color_roles.txt", "r") as file:
        rst_prolog += file.read()
        rst_prolog += "\n"

with open("roles/custom_roles.txt", "r") as file:
    rst_prolog += file.read()
    rst_prolog += "\n"


rst_prolog += "\n" + "\n".join(rst_prolog_lst)

# set if figures can be referenced as numers. Defalut is False
numfig = True

# set is todo's will show up.
# a master list of todo's will be on bottom of main page.
# https://www.sphinx-doc.org/en/master/usage/extensions/todo.html#module-sphinx.ext.todo
todo_include_todos = False

# region sphinx.ext.extlinks – Markup to shorten external links
# https://documentation.help/Sphinx/extlinks.html
# extlinks = {}

# region intersphinx
# intersphinx_mapping = {"odevgui": ("https://somedomain.com", None)}

# endregion intersphinx
```

### Relevant log output

```shell
Failed to handle request 810 textDocument/documentLink partial_result_token=None work_done_token=None text_document=TextDocumentIdentifier(uri='file:///d%3A/Users/bigby/Projects/python/win/rst-scratch/docs/scratch/go_figure_https.rst')
Traceback (most recent call last):
  File "d:\Users\bigby\Projects\python\win\rst-scratch\.venv\lib\site-packages\pygls\protocol.py", line 337, in _handle_request
    self._execute_request(msg_id, handler, params)
  File "d:\Users\bigby\Projects\python\win\rst-scratch\.venv\lib\site-packages\pygls\protocol.py", line 260, in _execute_request
    method_name, method_type, msg_id, handler(params))
  File "d:\Users\bigby\Projects\python\win\rst-scratch\.venv\lib\site-packages\esbonio\lsp\__init__.py", line 357, in on_document_link
    links += feature.document_link(context) or []
  File "d:\Users\bigby\Projects\python\win\rst-scratch\.venv\lib\site-packages\esbonio\lsp\directives.py", line 768, in document_link
    result = feature.resolve_argument_link(
  File "d:\Users\bigby\Projects\python\win\rst-scratch\.venv\lib\site-packages\esbonio\lsp\sphinx\images.py", line 81, in resolve_link
    return self.resolve_path(context.doc, argument), None
  File "d:\Users\bigby\Projects\python\win\rst-scratch\.venv\lib\site-packages\esbonio\lsp\sphinx\images.py", line 98, in resolve_path
    fpath = (basedir / argument).resolve()
  File "C:\ProgramData\Miniconda3\lib\pathlib.py", line 1215, in resolve
    s = self._flavour.resolve(self, strict=strict)
  File "C:\ProgramData\Miniconda3\lib\pathlib.py", line 215, in resolve
    s = self._ext_to_normal(_getfinalpathname(s))
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'd:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\docs\\scratch\\https:\\user-images.githubusercontent.com\\4193389\\205182487-b1796a72-ec04-4bdc-8a8c-26acdf72039e.png'
[Error - 10:32:47 AM] Request textDocument/documentLink failed.
  Message: OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'd:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\docs\\scratch\\https:\\user-images.githubusercontent.com\\4193389\\205182487-b1796a72-ec04-4bdc-8a8c-26acdf72039e.png'
  Code: -32602 
{'traceback': ['  File "d:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\.venv\\lib\\site-packages\\pygls\\protocol.py", line 337, in _handle_request\n    self._execute_request(msg_id, handler, params)\n', '  File "d:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\.venv\\lib\\site-packages\\pygls\\protocol.py", line 260, in _execute_request\n    method_name, method_type, msg_id, handler(params))\n', '  File "d:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\.venv\\lib\\site-packages\\esbonio\\lsp\\__init__.py", line 357, in on_document_link\n    links += feature.document_link(context) or []\n', '  File "d:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\.venv\\lib\\site-packages\\esbonio\\lsp\\directives.py", line 768, in document_link\n    result = feature.resolve_argument_link(\n', '  File "d:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\.venv\\lib\\site-packages\\esbonio\\lsp\\sphinx\\images.py", line 81, in resolve_link\n    return self.resolve_path(context.doc, argument), None\n', '  File "d:\\Users\\bigby\\Projects\\python\\win\\rst-scratch\\.venv\\lib\\site-packages\\esbonio\\lsp\\sphinx\\images.py", line 98, in resolve_path\n    fpath = (basedir / argument).resolve()\n', '  File "C:\\ProgramData\\Miniconda3\\lib\\pathlib.py", line 1215, in resolve\n    s = self._flavour.resolve(self, strict=strict)\n', '  File "C:\\ProgramData\\Miniconda3\\lib\\pathlib.py", line 215, in resolve\n    s = self._ext_to_normal(_getfinalpathname(s))\n']}
```

**Comment by alcarney** [comment]
Looks like the language server is not handling the error correctly, I've opened swyddfa/esbonio#506 to track this

**Comment by alcarney** [comment]
The fix for this has just been released in `v0.15.0` of `esbonio`

You can upgrade by running 
```
> python -m pip install --upgrade esbonio
```

in your project's Python env. Let me know if the issue persists

**Comment by Amourspirit** [comment]
Most Excellent! I just upgraded and all seems well.
